prow/tide/tide.go

```diff
@@ -276,57 +283,130 @@ func (c *Controller) pickBatch(sp subpool) ([]pullRequest, error) {
 	return res, nil
 }
 
-func (c *Controller) syncSubpool(ctx context.Context, sp subpool) error {
-	c.log.Infof("%s/%s %s: %d PRs, %d PJs.", sp.org, sp.repo, sp.branch, len(sp.prs), len(sp.pjs))
-	var presubmits []string
+func (c *Controller) mergePRs(sp subpool, prs []pullRequest) error {
+	for _, pr := range prs {
+		if err := c.ghc.Merge(sp.org, sp.repo, int(pr.Number), github.MergeDetails{
+			SHA: string(pr.HeadRef.Target.OID),
+		}); err != nil {
+			if _, ok := err.(github.ModifiedHeadError); ok {
+				// This is a possible source of incorrect behavior. If someone
+				// modifies their PR as we try to merge it in a batch then we
+				// end up in an untested state. This is unlikely to cause any
+				// real problems.
+				c.Logger.WithError(err).Info("Merge failed: PR was modified.")
+			} else if _, ok = err.(github.UnmergablePRError); ok {
+				c.Logger.WithError(err).Warning("Merge failed: PR is unmergable. How did it pass tests?!")
+			} else {
+				return err
```

Comment:
What kind of error this can be?

Reply:
If GitHub is down then we can't really recover properly. Better to just cut out here and retry on the next sync loop. I covered the two cases from which we can recover above.

Reply:
ack